### ISSUE TITLE
Optional Data and localized provider call, parse error

### ISSUE THREAD

**Issue by sylvaincombes** [nelmio/alice#956]
Hi, thank you for your project

When using optional data with localized syntax, I face a situation with a parse error :

```yml
App\Entity\Registration:
  en_{1..10}:
    address2: '50%? <en_US:streetAddress()>'
    ...
```
I guess it's the : from the \<locale:provider()\> ?

Here is command line output :

```
2018-10-01T11:12:20+02:00 [error] Error thrown while running command "hautelook:fixtures:load --env=test -q". Message: "An error occurred while denormalizing the fixture "en_{1..10}" (App\Entity\Registration): Invalid token "50%? <aliceTokenizedFunction(FUNCTION_START__en_US:streetAddress__IDENTITY_OR_FUNCTION_END)>" found."

In TolerantFixtureDenormalizer.php line 68:
                                                                                                                                                                                                                 
  An error occurred while denormalizing the fixture "en_{1..10}" (App\Entity\Registration): Invalid token "50%? <aliceTokenizedFunction(FUNCTION_START__en_US:streetAddress__IDENTITY_OR_FUNCTION_END)>" found.  
                                                                                                                                                                                                                 

In InvalidArgumentExceptionFactory.php line 125:
                                                                                                                       
  Invalid token "50%? <aliceTokenizedFunction(FUNCTION_START__en_US:streetAddress__IDENTITY_OR_FUNCTION_END)>" found.  
```

Have you got a solution for this situation, or maybe I missed something ?

Thanks

**Comment by theofidry** [comment]
the syntax should work so I would say it's a limitation of the current lexer/parser which can be solved in 3 ways:

- use a faker provider as a workaround: you can handle the percentage of chances there as well
- add a failing test case to nelmio/alice and try to solve it, I would recommend to look at Lexer and ParserIntegrationTestCase
- help with the real fix: #712

**Comment by theofidry** [comment]
Closing as the real fix IMO is #712 at that point
